src/backend/api/meals.js

```diff
@@ -76,4 +62,154 @@ router.delete("/:id", async (req, res) => {
   }
 });
 
+// toLowercaseMiddleware is used to parse reqest query to lowercase before passing them to the route handler function
+// after that function will called next method to pass control to the next middleware or route handler.
+// make sure that you add middleware function before route handler to ensure that all query parameters are converted to
+// lowercase before being used in any conditions.
+function toLowerCaseMiddleware(req, res, next) {
+  for (const key in req.query) {
+    if (typeof req.query[key] === "string") {
+      req.query[key] = req.query[key].toLowerCase();
+    }
+  }
+  next();
+}
+
+// Get request query
+router.get("/", toLowerCaseMiddleware, async (req, res) => {
+  const query = req.query;
+  let mealQuery = db("meal");
+  const datePattern = /^\d{4}-\d{2}-\d{2}$/;
+  let {
+    maxPrice,
+    availableReservations,
+    title,
+    dateAfter,
+    dateBefore,
+    limit,
+    sortKey,
+    sortDir,
+  } = req.query;
+
+  // MaxPrice
+  if ("maxPrice" in query) {
+    maxPrice = parseInt(maxPrice); // convert string number to number
+    if (!isNaN(maxPrice)) {
+      mealQuery = mealQuery.where("price", "<=", maxPrice);
+    } else {
+      return res.status(400).json({ Error: "Please fill in a valid number" });
+    }
+  }
+
+  // AvailableReservations
+  if ("availableReservations" in query) {
+    if (availableReservations === "true") {
+      mealQuery = mealQuery
+        .select(
+          "meal.title",
+          db.raw(
+            "(meal.max_reservation -  sum(reservation.number_of_guests)) as available_slot"
```

Comment:
There is some code repeated in both if and if-else block. Some of that code can be re-used to respect DRY (Do Not Repeat) principle